File: src/lib/presentation.ts
export interface Presentation {
  title: string;
  author?: string;
  slides: Slide[];
}

export type SlideLayout = "title" | "content" | "section" | "blank";

export interface Slide {
  id: string;
  layout: SlideLayout;
  title?: string;
  subtitle?: string;
  elements: SlideElement[];
}

export type ElementType = "text" | "image" | "bullet_list";

export interface SlideElement {
  type: ElementType;
  content: string | string[]; // string[] for bullet_list
  position?: {
    x: number; // percentage 0-100
    y: number;
    w: number;
    h: number;
  };
}

File: src-tauri/src/presentation.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Presentation {
    pub title: String,
    pub author: Option<String>,
    pub slides: Vec<Slide>,
}

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum SlideLayout {
    Title,
    Content,
    Section,
    Blank,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct Slide {
    pub id: String,
    pub layout: SlideLayout,
    pub title: Option<String>,
    pub subtitle: Option<String>,
    pub elements: Vec<SlideElement>,
}

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum ElementType {
    Text,
    Image,
    BulletList,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct SlideElement {
    #[serde(rename = "type")]
    pub element_type: ElementType,
    pub content: ElementContent,
    pub position: Option<ElementPosition>,
}

#[derive(Debug, Serialize, Deserialize)]
#[serde(untagged)]
pub enum ElementContent {
    Text(String),
    Bullets(Vec<String>),
}

#[derive(Debug, Serialize, Deserialize)]
pub struct ElementPosition {
    pub x: f64,
    pub y: f64,
    pub w: f64,
    pub h: f64,
}
